.circleci/config.yml

```diff
@@ -390,7 +390,15 @@ jobs:
             command: |
               bash -c "$PACKCHECK $BUILD" || exit 1
               count=$(find . -name "*.hs" -exec grep -H '\ $' {} \; | tee /dev/tty | wc -l)
-              exit $count
+              if [ $count > 0 ]
+              then exit 1
+              fi
+              git diff > tmp_len
```

Comment:
This won't work. The output will always be empty as there is nothing changed in the checked out tree. You need to do git diff between the current branch HEAD and the target branch in which the PR is to be merged. The current branch and target branch info should be available in some env vars.

Also you could just directly pipe the output of diff to the command below rather than putting it in a file first.

Reply:
In github, `${{ github.event.pull_request.head.sha }}` is the pull request base commit.
```
git diff ${{ github.event.pull_request.head.sha }} HEAD 
```
will work for all pull requests as expected.

In circleci, I am unable to immediately find the env var that has the branch or the commit being compared to.
Some interesting env vars are,
- `CIRCLE_BRANCH`: Current branch name
- `CIRCLE_SHA1`: Current checkout latest hash

A trivial way if the merge branch is always origin/master is:
```
git diff origin/master HEAD 
```

You can probably use the github CI.
```
git diff ${{ github.event.pull_request.head.sha }} HEAD
```
should work as expected.

`dit diff` might produce lines that might be over 80 that we have no control of.
These might be the git info messages, separators, etc.
We need to filter only lines that start with a `+`.
We can probably solve the whole task with `sed`.